website/docs/topics/long_contexts.md

```diff
@@ -0,0 +1,305 @@
+# Handling Long Context Conversations with Transform Messages
+
+Why do we need to handle long contexts? The problem arises from several constraints and requirements:
+
+1. Token limits: LLMs have token limits that restrict the amount of textual data they can process. If we exceed these limits, we may encounter errors or incur additional costs. By preprocessing the chat history, we can ensure that we stay within the acceptable token range.
+
+2. Context relevance: As conversations progress, retaining the entire chat history may become less relevant or even counterproductive. Keeping only the most recent and pertinent messages can help the LLMs focus on the most crucial context, leading to more accurate and relevant responses.
+
+3. Efficiency: Processing long contexts can consume more computational resources, leading to slower response times.
+
+## Transform Messages Capability
+
+The `TransformMessages` capability is designed to modify incoming messages before they are processed by the LLM agent. This can include limiting the number of messages, truncating messages to meet token limits, and more.
+
+### Installation and Setup
+
+````{=mdx}
+:::info Requirements
+Install `pyautogen`:
+```bash
+pip install pyautogen
+```
+
+For more information, please refer to the [installation guide](/docs/installation/).
+:::
+````
+
+```python
+import os
+import pprint
+import copy
+import re
+
+import autogen
+from autogen.agentchat.contrib.capabilities import transform_messages, transforms
+from typing import Dict, List
+
+config_list = autogen.config_list_from_json(
```

Comment:
Instead of using OAI_CONFIG_LIST, we can use a raw Python list of dictionaries and get the api key from environment. This is to simplify the setup users need to go through.